Fixed color model

sad-xu opened the pull request:
### Issues
<!-- Each pull request needs to be related to an issue, mention it here below -->
Fixes (#766)

### Proposed changes
<!-- Describe the proposed changes -->
The color settings in `display.resources.` are incorrect
See https://codepen.io/sad-xu/pen/RwjyQey
Because the color resolution is wrong
The original test case is only applicable to the #000 case 😂
Please check it

### Checklist
- [x] I consent that this change becomes part of alphaTab under it's current or any future open source license
- [x] Changes are implemented
- [ ] Existing builds tests pass
- [x] New tests were added

## Further details
- [ ] This is a breaking change
- [ ] This change will require update of the documentation/website

Danielku15 commented:
Can you first file a bug before providing a fix? Please be sure to test all platforms on your system. The proposed changes will not compile on C# and Kotlin.

Danielku15 commented on src/model/Color.ts:
```diff
@@ -27,7 +27,7 @@ export class Color {
                 ModelUtils.toHexString(this.b, 2);
         } else {
-            this.rgba = `rgba(${this.r},${this.g},${this.b},${this.a / 255.0})`;
+            this.rgba = `rgba(${this.r},${this.g},${this.b},${+(this.a / 255.0).toFixed(2)})`;
```
While this is valid code in Javascript/Typescript, the C# and Kotlin compilation will fail because there is by today no `ToFixed` extension method defined:
https://github.com/CoderLine/alphaTab/runs/5294838093?check_suite_focus=true#step:6:90
https://github.com/CoderLine/alphaTab/runs/5294837733?check_suite_focus=true#step:7:113

You can learn more about the toolchain here: https://alphatab.net/docs/contributing#development-environment

You can either extend all the platforms to solve this issue (introduce extension methods) or maybe this change is not actually required and could be reverted. It's not looking very nice to have that many decimals but technically I guess it doesn't hurt.

Danielku15 reviewed (changes requested):
Thanks for your contribution. There are some small things open before I can really accept this PR: 

1. For the sake of house-keeping, please create an issue describing the problem using the bug template and refer to it in this PR. 
2. Web Tests seem to be failing due to some casing problems: https://github.com/CoderLine/alphaTab/runs/5294838000?check_suite_focus=true#step:6:908
3. `ToFixed` is not available in C# and Kotlin (see comment below). 
4. # symbols in the test names will likely cause compilation issues in C# and Kotlin as we use them for method names. (see comment below)
5.

sad-xu commented:
I created the issue #766 
I removed `toFixed` and modified some test case
Only test passed  Web because I don't have the environment of  C# and Kotlin
Please help me check it 🙇

Danielku15 commented:
I pushed some improvements to develop which should fix some issues in the test method name handling. If you update your branch from the new develop on the upstream the errors should disappear.

Danielku15 reviewed (changes requested):
Please update your branch with the latest state from develop to make some errors disappear.

sad-xu commented:
I updated the branch 
Is there something wrong with my code? 😥

Danielku15 commented:
@sad-xu Generally the change is fine, but as described in [the toolchain](https://alphatab.net/docs/contributing#toolchain) page, the TypeScript code is translated to C# and Kotlin automatically. 

Your code like: 
```ts
    it('fromJson-#RGB', () => {
    // ...
    });
``` 

will result in C# code like: 
```cs
   [TestMethod]
   public void FromJson#RGB() { ... }
```

the `#` sign in the test names will cause problems when translated because you cannot use `#` in a method name in C# or Kotlin. I added some improvements to the toolchain to automatically replace such characters. The alternative would have been to simply name the methods differently like writing `fromJson-HashRgb` directly, but with my improvement your code should automatically work (in best case). 😉

Danielku15 commented:
Merged in  [d2f40af](https://github.com/CoderLine/alphaTab/commit/d2f40af6098e62c95be7fe0f34c6fd837e316a41)
there were some tricky pitfalls this change ran into. It uses some JS language features which weren't used until now in the code and therefore the C#/Kotlin toolchain wasn't prepared yet (indexer on strings, numeric literals assigned to unknown etc.). I had to improve some parts on my cross compilation toolchain to include these cases and now they are working. 

I had to merge it separately because I couldn't update this PR. 😉 

Thanks for your contribution. 🎉
